datafusion/core/src/physical_plan/stream.rs

```diff
@@ -73,3 +74,45 @@ impl RecordBatchStream for RecordBatchReceiverStream {
         self.schema.clone()
     }
 }
+
+/// Combines a [`BoxStream`] with [`SchemaRef`] implementing
+/// [`RecordBatchStream`] for the combination
+pub(crate) struct RecordBatchBoxStream {
```

Comment:
I'm not a massive fan of this, as it results in double dyn-dispatch, but it is probably acceptable until such a time as we rework ExecutionPlan.

I've put this alongside RecordBatchReceiverStream as it is likely generally useful, but I've kept it pub(crate) as I'm not sure how long this interface will be around for